internal/services/kusto/kusto_cosmosdb_data_connection_resource.go

```diff
@@ -0,0 +1,241 @@
+package kusto
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/cosmosdb/2023-04-15/cosmosdb"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/kusto/2022-12-29/databases"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/kusto/2022-12-29/dataconnections"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/kusto/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type CosmosDBDataConnectionModel struct {
+	Name                string `tfschema:"name"`
+	ResourceGroupName   string `tfschema:"resource_group_name"`
+	Location            string `tfschema:"location"`
+	CosmosDbContainerId string `tfschema:"cosmosdb_container_id"`
+	DatabaseId          string `tfschema:"kusto_database_id"`
+	ManagedIdentityId   string `tfschema:"managed_identity_id"`
+	MappingRuleName     string `tfschema:"mapping_rule_name"`
+	RetrievalStartDate  string `tfschema:"retrieval_start_date"`
+	TableName           string `tfschema:"table_name"`
+}
+
+var _ sdk.Resource = CosmosDBDataConnectionResource{}
+
+type CosmosDBDataConnectionResource struct{}
+
+func (r CosmosDBDataConnectionResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.DataConnectionName,
+		},
+		"resource_group_name": commonschema.ResourceGroupName(),
```

Comment:
This shouldn't be required anymore since we can infer this from `kusto_database_id`
```suggestion
```